src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -187,7 +187,7 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                         {
                             foreach (BoundPropertySubpattern subpattern in rp.Properties)
                             {
-                                if (subpattern.Member is BoundPropertySubpatternMember member)
+                                if (subpattern.Member is BoundPropertySubpatternMember { HasErrors: false } member)
```

Comment:
It feels like a better solution would be checking if the symbol of the pattern is a valid symbol kind for having a slot. In this case, the symbol is a NamedTypeSymbol, and thus it shouldn't have a slot. Bailing out if there are any errors feels a bit too broad.

Reply:
I had a specific check as you suggested in earlier iteration but felt that a broader check would be more defensive ;-)

Reply:
I think the broader check is both too broad and incorrect, as there's nothing directly connecting the symbol being valid to the HasErrors flag.

Reply:
With the broader check, only binding has to know what is correct. If the symbols are invalid, then it flags the error. So I don't understand what you mean by incorrect.

Reply:
I mean that there is no guarantee that, just because HasErrors is false, that the pattern symbol can have a slot. HasErrors is a flag intended for suppressing later errors. Having it affect nullable analysis this way is not the intended purpose.

Reply:
As I explained above: bad symbol implies HasError, so the converse holds and can be depended on: !HasErrors implies valid symbol.
I don't think it's worth holding off the PR on a nitpick, so I made the change.

Reply:
Thanks. I did think it was worth holding off approval for :).